core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystemCache.java

```diff
@@ -0,0 +1,231 @@
+			for (Folder localFolder : folders) {
+				registerFolder(mailbox, localFolder);
+			}
+
+			mailboxesList.add(mailbox);
+		}
+	}
+
+	public FolderId getBaseFolderId(String emailAddress, String baseFolderName, ExchangeService service) throws FileSystemException {
+		FolderId basefolderId;
+
+		log.debug("searching inbox");
+		FolderId inboxId;
+		if (StringUtils.isNotEmpty(emailAddress)) {
+			Mailbox mailbox = new Mailbox(emailAddress);
+			inboxId = new FolderId(WellKnownFolderName.Inbox, mailbox);
+		} else {
+			inboxId = new FolderId(WellKnownFolderName.Inbox);
+		}
+		log.debug("determined inbox ["+inboxId+"] foldername ["+inboxId.getFolderName()+"]");
+
+		if (StringUtils.isNotEmpty(baseFolderName)) {
+			try {
+				basefolderId=findFolder(service,inboxId,baseFolderName);
+			} catch (Exception e) {
+				throw new FileSystemException("Could not find baseFolder ["+baseFolderName+"] as subfolder of ["+inboxId.getFolderName()+"]", e);
+			}
+			if (basefolderId==null) {
+				log.debug("Could not get baseFolder ["+baseFolderName+"] as subfolder of ["+inboxId.getFolderName()+"]");
+				basefolderId=findFolder(service,null,baseFolderName);
+			}
+			if (basefolderId==null) {
+				throw new FileSystemException("Could not find baseFolder ["+baseFolderName+"]");
+			}
+		} else {
+			basefolderId=inboxId;
+		}
+
+		return basefolderId;
+	}
+
+	/**
+	 * Puts a Folder object in cache for specified mailbox.
+	 *
+	 * @param mailbox - The name of the mailbox to cache.
+	 * @param folder - The Folder object to store in cache.
+	 */
+	public void registerFolder(String mailbox, Folder folder) throws Exception {
+		String folderName = folder.getDisplayName();
+		String key = mailbox + folderName;
+		if(!folders.containsKey(key)){
+			log.debug("Creating a local cache of folder ["+folderName+"] for ["+mailbox+"] under key ["+key+"].");
+
+			folders.put(key, folder);
+		}
+	}
+
+	public void removeFolder(String mailbox, Folder folder) throws Exception {
+		String folderName = folder.getDisplayName();
+		String key = mailbox + folderName;
+		if(folders.containsKey(key)){
+			log.debug("Removing folder ["+folderName+"] for ["+mailbox+"] under key ["+key+"] from cache.");
+
+			folders.remove(key);
+		}
+	}
+
+	/**
+	 * Standard method to retrieve a list of Folder objects within a parent Folder.
+	 *
+	 * @param service - An instance of Exchange Service to use to make the call
+	 * @param parentFolderId - The ID of the parent folder to check for sub Folders.
+	 *
+	 * @return ArrayList<Folder> - The list of found Folder objects within parent folder.
+	 */
+	private ArrayList<Folder> findFolders(ExchangeService service, FolderId parentFolderId, int folderViewCount) throws Exception {
+		return service.findFolders(parentFolderId, new FolderView(folderViewCount)).getFolders();
+	}
+
+	private FolderId findFolder(ExchangeService service, FolderId baseFolderId, String folderName) throws FileSystemException {
```

Comment:
Deze method is vrijwel 1:1 verplaatst uit ExchangeFileSystem. Graag daar weer terug plaatsen, zodat duidelijk is wat de aanpassingen zijn. Hij heeft voor zover ik kan zien geen afhankelijkheden met ExchangeFileSystemCache